### PR TITLE
MCO-594: Remove Pivot Command & Container File Edit

### PR DESCRIPTION
**Pull request by isabella-janssen**
**- What I did**
- Removed the `pivot.go` cmd file & unused `EtcPivotFile` constant
- Updated container file build user creation to first check if the user exists (credit: @cheesesashimi)

**- How to verify it**
Attempt to run pivot on any node, it should error that the command does not exist.
```
sh-5.1# /run/bin/machine-config-daemon pivot test
Error: unknown command "pivot" for "machine-config-daemon"
```

**- Description for the changelog**
[MCO-594](https://issues.redhat.com//browse/MCO-594): Removes non-functional pivot command

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -44,7 +44,8 @@ RUN --mount=type=cache,target=/var/cache/dnf,z \
     # Create the build user which will be used for doing OS image builds. We
     # use the username "build" and the uid 1000 since this matches what is in
     # the official Buildah image.
-    useradd --uid 1000 build
+    # Conditional checks if "build" user does not exist before adding user.
+    if ! id -u "build" >/dev/null 2>&1; then useradd --uid 1000 build; fi
```
Comment (dkhater-redhat) [inline Dockerfile:48]: is this work related to the pivot command? if not it's totally cool but I would maybe update the PR title to reflect this addition.

Comment (dkhater-redhat) [inline Dockerfile:48]: is this work related to the pivot command? if not it's totally cool but I would maybe update the PR title to reflect this addition.

**Comment by dkhater-redhat** [comment]
everything looks good! I would squash those commits 🥪

File: Dockerfile (hunk 1 above)
Comment (isabella-janssen) [reply Dockerfile:48]: No, it's not part of removing the pivot command, just a nice enhancement from Zack. I went ahead and updated the PR title!

**Review by yuqi-zhang — COMMENTED** [review]
Two suggestions inline, also, it seems that we have this service:
https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/units/pivot.service.yaml

I think that drop-in definition is for a non-existant pivot service. Could you also check whether anything is shipping that in RHCOS and if not, remove that service template?

Other than those, all the other pivot* named functionality in the MCO are for something else it looks like, so hopefully that's all the remaining changes :) 

@djoshy do you remember if we need to update any documentation around this outside of the repo?

File: docs/HACKING.md [hunk 2]
```diff
@@ -299,10 +299,6 @@ code related to upgrades.  For this, see https://github.com/coreos/coreos-assemb
 But let's assume you have a custom container and have pushed to a registry, for
 this example `quay.io/example/rhel-coreos:latest`.
 
-Once you have an oscontainer, you can again use `oc debug node/` and  `pivot` to directly switch
```
[the remark this answers is not in the document]
Comment (djoshy) [reply docs/HACKING.md:302]: Hmm, correct me if I am wrong, but I think this references https://github.com/openshift/pivot before it was folded into the MCO, so its not directly related to removing the daemon command :thinking:

**Comment by djoshy** [comment]
> @djoshy do you remember if we need to update any documentation around this outside of the repo?
Not that I can think of, we removed this "support path" from the KB article when we first deprecated it, which seemed to be the main source of how customers ran into this: https://access.redhat.com/solutions/5598401

File: docs/HACKING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuqi-zhang) [reply docs/HACKING.md:302]: Right, this is in a sense even more outdated. I think we should update the doc still, but I guess it doesn't have to be part of this PR

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (isabella-janssen) [reply Dockerfile:48]: Containerfile edit has been split into a separate commit.

**Comment by djoshy** [comment]
/lgtm
/approve

I don't think we'll need QE review here..since we are removing deprecated functionality, we should be covered by our e2es.
